gnnflow/distributed/kvstore.py

```diff
@@ -53,27 +103,133 @@ class KVStoreClient:
     It is used by the trainer to push/pull tensors to/from the KVStore servers.
     """
 
-    def __init__(self, partition_table: torch.Tensor):
+    def __init__(self, partition_table: torch.Tensor,
+                 num_partitions: int,
+                 num_workers_per_machine: int,
+                 local_rank: int):
         self._partition_table = partition_table
+        self._num_partitions = num_partitions
+        self._num_workers_per_machine = num_workers_per_machine
+        self._local_rank = local_rank
 
-    def push(self, keys: torch.Tensor, tensors: List[torch.Tensor]):
+    def push(self, keys: torch.Tensor, tensors: List[torch.Tensor], mode: str, nid: Optional[torch.Tensor] = None):
```

Comment:
I don't understand the API here. In the following code, `tensors` is just a tensor, not a list of tensors.

Reply:
Yes, in fact it's a tensor whose shape[0] equals to len(keys). Not a list of tensors

Reply:
Then update  the API.